core/variant/variant_parser.cpp

```diff
@@ -1214,6 +1214,7 @@ Error VariantParser::_parse_tag(Token &token, Stream *p_stream, int &line, Strin
 			r_tag.name += String::chr(c);
 		}
 
+		r_tag.name.parse_utf8(r_tag.name.ascii(true).get_data());
```

Comment:
Won't that fail on anything outside of extended ASCII?

CC @bruvzg

Reply:
This can be used with `StreamFile` which return UTF-8 chars, and with `StreamString` which return UTF-32 chars.

It will work only if wrapped in `if (p_stream->is_utf8()) { ... }`, but probably should be something like:
```c++
if (p_stream->is_utf8()) {
  while (true) {
    char32_t c = p_stream->get_char();
    if (p_stream->is_eof()) {
      r_err_str = "Unexpected EOF while parsing simple tag";
      return ERR_PARSE_ERROR;
    }
    if (c == ']') {
      break;
    }
    r_tag.name += String::chr(c);
  }
} else {
  CharString cs;
  while (true) {
    char c = p_stream->get_char();
    if (p_stream->is_eof()) {
      r_err_str = "Unexpected EOF while parsing simple tag";
      return ERR_PARSE_ERROR;
    }
    if (c == ']') {
      break;
    }
    cs += c;
  }
  r_tag.name.parse_utf8(cs.get_data(), cs.length());
}

```

Reply:
I tried it with extended latin and with Japanese characters and it worked.

Reply:
It will work if `p_stream` is `StreamFile`, but won't with `StreamString`.

```c++
if (p_stream->is_utf8()) {
   r_tag.name.parse_utf8(r_tag.name.ascii(true).get_data());
}
```
With a check, it will work in both cases.

Reply:
Which with glyphs does it currently not work?

Reply:
With `StreamString` it should fail with any character in 0xC0-0xF7 range.

Reply:
Not sure if this function is ever used with `StreamString`, but simple test code fails as expected:
```
	char32_t aa = U'Â';
	String a = "A" + String::chr(aa) + "T";
	String b;
	b.parse_utf8(a.ascii(true).get_data());
	printf("%s %s\n", a.utf8().get_data(), b.utf8().get_data());
```

```
Unicode parsing error: invalid utf8. Is the string valid UTF-8?
AÂT A
```

With `is_utf8` check it will always work, but it's using wrong container (`String` instead of `CharString`) to store UTF-8 data, which leads to pretty misleading.